pino.js

```diff
@@ -52,48 +63,48 @@ function streamIsBlockable (s) {
 }
 
 function pino (opts, stream) {
-  if (opts && (opts.writable || opts._writableState)) {
-    stream = opts
-    opts = null
+  var iopts = opts
+  var istream = stream
+  if (iopts && (iopts.writable || iopts._writableState)) {
+    istream = iopts
+    iopts = defaultOptions
   }
-  stream = stream || process.stdout
-  opts = opts || {}
-  var timestamp = (opts.hasOwnProperty('timestamp')) ? opts.timestamp : true
-  var slowtime = opts.slowtime
-  var safe = opts.safe !== false
-  var stringify = safe ? stringifySafe : JSON.stringify
-  var formatOpts = safe ? null : {lowres: true}
-  var name = opts.name
-  var level = opts.level || 'info'
-  var serializers = opts.serializers || {}
-  var end = ',"v":' + LOG_VERSION + '}\n'
-  var cache = !opts.extreme ? null : {
+  istream = istream || process.stdout
+  iopts = extend({}, defaultOptions, iopts)
+
+  // internal options
+  iopts.stringify = iopts.safe ? stringifySafe : JSON.stringify
+  iopts.formatOpts = iopts.safe ? null : {lowres: true}
+  iopts.end = ',"v":' + LOG_VERSION + '}\n'
+  iopts.cache = !iopts.extreme ? null : {
     size: 4096,
     buf: ''
   }
+  iopts.chindings = ''
 
-  if (opts.enabled === false) {
-    level = 'silent'
+  if (iopts.enabled === false) {
+    iopts.level = 'silent'
   }
 
-  var logger = new Pino(level, stream, serializers, stringify, end, name, timestamp, slowtime, '', cache, formatOpts)
-  if (cache) {
+  var logger = new Pino(iopts, istream)
+  if (iopts.cache) {
     // setImmediate is causing a very weird crash:
     //    Assertion failed: (cb_v->IsFunction()), function MakeCallback...
     // but setTimeout isn't *shrug*
     setTimeout(function () {
-      if (!streamIsBlockable(stream)) {
+      if (!streamIsBlockable(istream)) {
         logger.emit('error', new Error('stream must have a file descriptor in extreme mode'))
       }
     }, 100)
 
     onExit(function (code, evt) {
-      if (cache.buf) {
+      var buf = iopts.cache.buf
+      if (buf) {
         // We need to block the process exit long enough to flush the buffer
         // to the destination stream. We do that by forcing a synchronous
         // write directly to the stream's file descriptor.
-        var fd = (stream.fd) ? stream.fd : stream._handle.fd
-        require('fs').writeSync(fd, cache.buf)
+        var fd = (istream.fd) ? istream.fd : istream._handle.fd
+        require('fs').writeSync(fd, buf)
```

Comment:
I don't know where this came from the first time, but we shouldn't be requiring inline as a matter of principle, would you mind moving this out